src/Avalonia.Controls/Shapes/Shape.cs

```diff
@@ -254,7 +254,7 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
                     InvalidateMeasure();
                 }
                 
-                if (!Pen.TryModifyOrCreate(ref _strokePen, Stroke, StrokeThickness, StrokeDashArray, StrokeDashOffset, StrokeLineCap, StrokeJoin))
+                if (Pen.TryModifyOrCreate(ref _strokePen, Stroke, StrokeThickness, StrokeDashArray, StrokeDashOffset, StrokeLineCap, StrokeJoin))
```

Comment:
I think we should just call InvalidateVisual all the time here because otherwise setting the Stroke to null will not update the rendering

Reply:
The point of TryModifyOrCreate is to modify underlying Pen instance without invalidating whole shape. 
Or are you referring to something else?